Code/Framework/AzCore/AzCore/IO/Path/PathParser.inl

```diff
@@ -626,6 +626,46 @@ namespace AZ::IO::parser
         return string_view_pair{ srcView.substr(0, pos), srcView.substr(pos) };
     }
 
+    // path part decomposition
+    enum class PathPartKind : uint8_t
+    {
+        PK_None,
```

Comment:
The PK_ prefix stood out as odd. Bouncing around a code search and it doesn't seem like prefixing enum names is standard. I guess the enum existed before you got here, soooo... fine to leave it.

Reply:
Yeah, especially since this is an `enum class`.

Reply:
This code was moved from the bottom of the file to above the `CompareRelative` function code.
However, I was the one who wrote all of this code in the first place, so I added the enum prefix.
Though I may have gotten the naming scheme for the Path Part Kind from looking at the libstdc++ filesystem::path code

The coding standard however, does not recommend against using a prefixing a scoped enum.
It only recommends that an unscoped enum is prefixed: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#enums